src/targets/gpu/mlir.cpp

```diff
@@ -947,10 +947,12 @@ struct mlir_program
     std::string sym_name;
 };
 
-bool is_module_fusible(const module& m, const value& solution)
+bool is_module_fusible(const module& m, const context& migraphx_ctx, const value& solution)
 {
     mlir_program mp;
+    mp.set_gpu_properties(migraphx_ctx);
```

Comment:
Does gpu properties need to be set for the high level pipeline?

Reply:
Yes otherwise MLIR prints warnings like 
`Warning: unknown architecture, falling back to defaults:`

Reply:
But is that a problem for `is_module_fusible`?

Reply:
`set_gpu_properties` is only setting the `arch` on the MLIR function. MLIR can potentially run different route while running high level backend pipeline. 

Also we don't want warnings that can be fixed.

Reply:
I assume the high level pipeline is target independent so it seems unnecessary, but this would be a better question for @sjw36 @krzysz00 @manupak to answer.

Reply:
MLIR team can answer it, but i don't see why this issue would be a blocker for approval. I don't see any harm in having a call to `set_target_properties()` and setting `arch` attribute.  arch is set later too when compiling MLIR module.

Reply:
> I assume the high level pipeline is target independent so it seems unnecessary.

In a practical sense yes, but we do extract features from arch in the high-level; so that further lowerings (low-level) only look at a bag of features that is there on the arch.